### PR TITLE
Add ADU support for L4plus

### PR DESCRIPTION
**Pull request by danewalton**
thanks to @vaavva we pretty much get this for free :)

### CODE REVIEW

File: demos/projects/ST/b-l4s5i-iot01a/config/demo_config.h [hunk 1]
```diff
@@ -510,4 +510,16 @@ static unsigned char root_cert_array[] = {
  */
 #define WIFI_SECURITY_TYPE  WIFI_ECN_WPA2_PSK
 
+#ifdef democonfigENABLE_DPS_SAMPLE
+    #define democonfigCHUNK_DOWNLOAD_SIZE    256
```
Comment (vaavva) [inline demos/projects/ST/b-l4s5i-iot01a/config/demo_config.h:514]: I need to go and check whether these numbers can be updated for the L475 sample now that the json buffer is gone, but does the L4plus have the same space constraints, or could these numbers potentially be bigger here as well? If I remember correctly, changing these to something too big causes a failure on build, so it's a pretty fast check!

Comment (danewalton) [reply demos/projects/ST/b-l4s5i-iot01a/config/demo_config.h:514]: Yea I think I was able to get north of 2k not using DPS (possibly 4k). Let me do some fudging on the numbers and see what we can handle now.

**Review by vaavva — APPROVED** [review]
This is awesome!! Thanks for getting this quick win :)

Here are the things I would double check if you haven't yet:
- azureiotflashL475_DOUBLE_WORD_SIZE - since we're setting this, I'd check and make sure it's the same for this board. azureiotflashL475_FLASH_ROW_SIZE would definitely cause failures with the image if it were wrong, but at the end of the bin file where we're programming by the double word it's mostly just writing 0s, so I could see this still working even if it's the wrong value.
- I would assume since these are defined in the ST HAL file, that it correctly figures out the board, but I just wanted to double check that it would since the flash platform file is in the L475 folder. FLASH_BASE and FLASH_BANK_SIZE

**Review by wduraes — APPROVED** [review]

File: demos/projects/ST/b-l4s5i-iot01a/CMakeLists.txt [hunk 2]
```diff
@@ -123,3 +125,60 @@ add_custom_command(TARGET ${PROJECT_NAME}-pnp
...
+    ${SOURCE_DIR}
+    ${SOURCE_DIR}/st_code)
+target_link_libraries(${PROJECT_NAME}-adu PRIVATE
+    FreeRTOS::Timers
+    FreeRTOS::Heap::5
+    FreeRTOS::ARM_CM4F
+    FreeRTOSPlus::Utilities::backoff_algorithm
+    FreeRTOSPlus::Utilities::logging
+    FreeRTOSPlus::ThirdParty::mbedtls
+    HAL::STM32::L4::RCC
```
Comment (ericwol-msft) [inline demos/projects/ST/b-l4s5i-iot01a/CMakeLists.txt:144]: could we group these as base L4 libs and only define them once?

**Review by ericwol-msft — APPROVED** [review]

**Review by ericwol-msft — APPROVED** [review]

File: demos/projects/ST/b-l475e-iot01a/port/azure_iot_flash_platform.c [hunk 3]
```diff
@@ -94,68 +94,23 @@ AzureIoTResult_t AzureIoTPlatform_WriteBlock( AzureADUImage_t * const pxAduImage
...
-                               FLASH_TYPEPROGRAM_FAST_AND_LAST : FLASH_TYPEPROGRAM_FAST, ( uint32_t ) pucNextWriteAddr, ( uint32_t ) pucNextReadAddr ) != HAL_OK )
+        if( HAL_FLASH_Program( FLASH_TYPEPROGRAM_DOUBLEWORD, ( uint32_t ) pucNextWriteAddr, ( uint64_t ) *( uint32_t * ) pucNextReadAddr | ( ( uint64_t ) *( uint32_t * ) ( pucNextReadAddr + 4 ) ) << 32 ) != HAL_OK )
         {
             /* Error occurred while writing data in Flash memory */
             xResult = eAzureIoTErrorFailed;
             break;
         }
 
-        pucNextWriteAddr += azureiotflashL475_FLASH_ROW_SIZE;
-        pucNextReadAddr += azureiotflashL475_FLASH_ROW_SIZE;
```
Comment (vaavva) [inline demos/projects/ST/b-l475e-iot01a/port/azure_iot_flash_platform.c:140]: I think the `azureiotflashL475_FLASH_ROW_SIZE` can be removed now as well!

File: demos/projects/ST/b-l4s5i-iot01a/config/demo_config.h (hunk 1 above)
Comment (danewalton) [reply demos/projects/ST/b-l4s5i-iot01a/config/demo_config.h]: Bumped to 4k with or without DPS

File: demos/projects/ST/b-l4s5i-iot01a/config/demo_config.h [hunk 4]
```diff
@@ -510,11 +510,7 @@ static unsigned char root_cert_array[] = {
  */
 #define WIFI_SECURITY_TYPE  WIFI_ECN_WPA2_PSK
 
-#ifdef democonfigENABLE_DPS_SAMPLE
-    #define democonfigCHUNK_DOWNLOAD_SIZE    256
-#else
-    #define democonfigCHUNK_DOWNLOAD_SIZE    1024
-#endif /* democonfigENABLE_DPS_SAMPLE */
+#define democonfigCHUNK_DOWNLOAD_SIZE    4096
```
Comment (vaavva) [inline demos/projects/ST/b-l4s5i-iot01a/config/demo_config.h:513]: woohoo!!

File: demos/projects/ST/b-l475e-iot01a/config/demo_config.h [hunk 5]
```diff
@@ -511,11 +511,7 @@ static unsigned char root_cert_array[] = {
  */
 #define WIFI_SECURITY_TYPE                   WIFI_ECN_WPA2_PSK
 
-#ifdef democonfigENABLE_DPS_SAMPLE
-    #define democonfigCHUNK_DOWNLOAD_SIZE    256
-#else
-    #define democonfigCHUNK_DOWNLOAD_SIZE    1024
-#endif /* democonfigENABLE_DPS_SAMPLE */
+#define democonfigCHUNK_DOWNLOAD_SIZE        2048
```
Comment (danewalton) [inline demos/projects/ST/b-l475e-iot01a/config/demo_config.h:514]: @vaavva updated this to be half of L4Plus but consistent between DPS and non

**Comment by danewalton** [comment]
Verified `A -> B -> A`

**Pull request merged**